rskj-core/src/main/java/co/rsk/peg/union/UnionBridgeSupportImpl.java

```diff
@@ -116,6 +116,53 @@ public Optional<Coin> getLockingCap() {
             storageProvider.getLockingCap(activations).orElse(constants.getInitialLockingCap()));
     }
 
+    @Override
+    public int increaseLockingCap(Transaction tx, Coin newCap) {
+        final String INCREASE_LOCKING_CAP_TAG = "increaseLockingCap";
+
+        if (isUnauthorizedCaller(tx, INCREASE_LOCKING_CAP_TAG)) {
+            return UnionResponseCode.UNAUTHORIZED_CALLER.getCode();
+        }
+
+        if (isInvalidLockingCap(newCap)) {
+            return UnionResponseCode.INVALID_VALUE.getCode();
+        }
+
+        storageProvider.setLockingCap(newCap);
+        logger.info("[{}] Union Locking Cap has been increased. New value: {}",
+            INCREASE_LOCKING_CAP_TAG, newCap.value);
+        return UnionResponseCode.SUCCESS.getCode();
+    }
+
+    private boolean isInvalidLockingCap(Coin newCap) {
+        Coin currentLockingCap = storageProvider.getLockingCap(activations)
+            .orElse(constants.getInitialLockingCap());
+
+        if (newCap.compareTo(currentLockingCap) < 1) {
+            logger.warn(
+                "[{}] Attempted value doesn't increase Union Locking Cap. Value attempted: {} . currentLockingCap: {}",
+                "isInvalidLockingCap", newCap.value, currentLockingCap.value);
+            return true;
+        }
+
+        Coin maxLockingCapIncreaseAllowed = currentLockingCap.multiply(
+            constants.getLockingCapIncrementsMultiplier());
+        if (newCap.compareTo(maxLockingCapIncreaseAllowed) > 0) {
+            logger.warn(
+                "[{}] Attempted value tries to increase Union Locking Cap above its limit. Value attempted: {} . maxLockingCapIncreasedAllowed: {}",
+                "isInvalidLockingCap", newCap.value, maxLockingCapIncreaseAllowed.value);
+            return true;
+        }
+
+        if (newCap.compareTo(constants.getMaxRbtc()) > 0) {
+            logger.warn(
+                "[{}] Attempted value tries to increase Union Locking Cap above the maximum RBTC value. Value attempted: {} . maxRbtc: {}",
+                "isInvalidLockingCap", newCap.value, constants.getMaxRbtc().value);
+            return true;
+        }
```

Comment:
Do we need this? What's the problem if locking cap goes above 21,000,000?

Reply:
No problem at all, it just makes no sense to set a locking cap that surpasses the max rbtc.

Reply:
Yeah but it add the complexity of having a new constant, new validations, new tests. Probably not worth it the way I see it

Reply:
Removed.